examples/main.cpp

```diff
@@ -107,20 +107,30 @@ static bool downscale_img_to_screen(sam_image_u8 &img, SDL_Window* window) {
 
 static bool load_image_from_file(const std::string & fname, sam_image_u8 & img) {
     int nx, ny, nc;
-    auto data = stbi_load(fname.c_str(), &nx, &ny, &nc, 3);
-    if (!data) {
-        fprintf(stderr, "%s: failed to load '%s'\n", __func__, fname.c_str());
+
+    // instead of directly loading the using stbi_load, lets first check if image is in supported format or not
+    int ok = stbi_info(filename, &nx, &ny, &nc);
```

Comment:
This doesn't compile. filename is undefined. You also added a couple of trailing tabs below.